js/src/hooks/useActionedCampaignsCache.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { LOCAL_STORAGE_KEYS, DAY_IN_SECONDS } from '~/constants';
+import localStorage from '~/utils/localStorage';
+
+const useActionedCampaignsCache = (
```

Comment:
This looks like a good local storage cache solution but currently seems either too specific to actioned campaigns to be used as a flexible local storage cache solution or else more flexible than it needs to be to store actioned raise budget recommendations.

Creating a flexible, general use cache solution is probably out of scope for what we need to create here, so I think this could be reduced to meet the specific use case we have here and eliminate the need for supporting a `storageKey` and `ttl` option.